### PR TITLE
Front: rolling stock editor: fix minor issues

### LINKED ISSUE #4510: front: rollingstockeditor: fix rollingstock editor scroll
### What happened?

if we scroll in the list of rolling stocks the entire page scrolls

### What did you expect to happen?

only the list of rolling stocks must scroll, the "add new rolling stock" button and the filters must stay on top of the page

### How can we reproduce it (as minimally and precisely as possible)?

go to rolling stock editor
try to scroll

### What operating system, browser and environment are you using?

- **Browser**: Firefox 114
- **OS**: MacOS Ventura
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

...

### LINKED ISSUE #4540: Rolling stock editor: Fix minor issues
### Description and goal

Quelques corrections mineures sont à apporter à l'éditeur de matériel roulant.

### Acceptance criteria

- [x] ajuster le comportement de l'image par défault.
- [x] ajuster le CSS de la liste des matériels roulants. #4510 
- [x] éviter le double appel pour récupérer un matériel roulant.
- [x] ajouter un message d'erreur global lorsqu'une erreur survient à la duplication.
- [x] ajouter une taille max aux images de matériel roulant pour ne pas avoir de boutons cachés lorsque celle-ci est trop grande.

Bonus: 
- [x] garder le formulaire ouvert, avec les champs renseignés en cas d'erreur lors de l'ajout / édition. (Cela évitera à l'utilisateur de tout recommencer)

### PR DESCRIPTION
**Pull request by SarahBellaha**
Closes #4510 
Closes #4540

### CODE REVIEW

**Review by Yohh — APPROVED** [review]
LGTM :muscle:

File: front/src/applications/rollingStockEditor/components/RollingStockEditorCard.tsx [hunk 1]
```diff
@@ -1,57 +1,52 @@
...
 
 type RollingStockEditorCardProps = {
+  id: number;
   isEditing: boolean;
-  setIsEditing: Dispatch<SetStateAction<boolean>>;
   rollingStock: RollingStock;
 };
 
 export default function RollingStockEditorCard({
+  id,
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorCard.tsx:17]: Why did you add this id ? You can get it from rollingStock.id no ? (maybe I miss something ?)

File: front/src/common/RollingStockSelector/RollingStock2Img.tsx [hunk 2]
```diff
@@ -27,19 +30,20 @@ const RollingStock2Img: React.FC<{ rollingStock: RollingStock | LightRollingStoc
...
-    // TODO : replace with null when images RTK query is ready in RollingStockEditor
-    <div className="defaultImageRSEditor">
-      <img src={blueTrain} alt="defaultImage" title="tchou tchou" />
-    </div>
-  );
+  useEffect(() => {
+    displayImage();
+  }, [imageUrl]);
+
+  return displayImage();
```
Comment (clarani) [inline front/src/common/RollingStockSelector/RollingStock2Img.tsx]: When a state variable changes, the component is automatically re-rendered ! So each time imageUrl will change, the LazyLoadImage will be re-rendered
```suggestion
  return <LazyLoadImage src={imageUrl || ''} alt={rollingStock?.name || 'defaultImg'} />;
```

File: front/src/common/RollingStockSelector/RollingStock2Img.tsx [hunk 3]
```diff
@@ -27,19 +30,20 @@ const RollingStock2Img: React.FC<{ rollingStock: RollingStock | LightRollingStoc
     }
   };
 
+  const displayImage = () => (
+    <LazyLoadImage src={imageUrl || ''} alt={rollingStock?.name || 'defaultImg'} />
+  );
+
```
Comment (clarani) [inline front/src/common/RollingStockSelector/RollingStock2Img.tsx]:
```suggestion
```

**Review by clarani — CHANGES_REQUESTED** [review]

File: front/src/common/RollingStockSelector/RollingStock2Img.tsx (hunk 3 above)
Comment (SarahBellaha) [reply front/src/common/RollingStockSelector/RollingStock2Img.tsx]: Done :)

File: front/src/common/RollingStockSelector/RollingStock2Img.tsx (hunk 2 above)
Comment (SarahBellaha) [reply front/src/common/RollingStockSelector/RollingStock2Img.tsx]: Done :)

File: front/src/applications/rollingStockEditor/components/RollingStockEditorCard.tsx (hunk 1 above)
Comment (SarahBellaha) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorCard.tsx:17]: When we take the id of the rollingstock passed in props, we retrieve the one called in **RollingStockEditor**. There is also a call to retrieve the rolling stock in **RollingStockCardDetail**. As we don't immediately have the new id of the `selectedRollingStock` called in **RollingStockEditor**, there's this double call that's made with the old id (deleted rollingstock) + the new id (opened one), which generates an error. To avoid this, I pass the id of the rollingStock that was opened instead.

**Review by clarani — APPROVED** [review]
LGTM !! :muscle:
